Mytinerary/Models/Event.m

```diff
@@ -28,6 +28,23 @@ + (nonnull NSString *)parseClassName {
  optional: event description, notes) */
 - (void) initNewEvent:(NSString *)title eventDescription:(NSString * _Nullable)eventDescription address:(NSString * _Nullable)address category:(NSString *)category startTime:(NSDate *)startTime endTime:(NSDate *)endTime notes:(NSString * _Nullable)notes withCompletion:(PFBooleanResultBlock)completion {
     
+    // make sure all required fields are filled, validate input
+    if ([title isEqualToString:@""]) {
+        NSError *error = [NSError errorWithDomain:@"Please add a title" code:1 userInfo:nil];
+        completion(nil, error);
+        return;
+    }
+    if ([startTime compare:endTime] != NSOrderedAscending) {
```

Comment:
you could probably auto change the start/end time if a new one is selected that conflicts with the currently selected one

like if you selected 6/1 as start time and selected 5/31 as end time, it would auto change the start time to 5/31